### ISSUE TITLE
Phantom crawls 

### ISSUE THREAD

**Issue by kbraak** [gbif/crawler#3]
#### Below is a detailed description of a phantom crawl attempt having no start date and that never comes to an end:**

[Mammals in MZNA-VERT: project CAS](http://www.gbif.org/dataset/b1b5147c-0ff8-4c3c-aff0-c5d78aff44db) was registered Mar 22, 2016 and republished again on Apr 14, 2016 according to its [version history](http://www.gbif.es/ipt/resource?r=mzna_vert_mast_cas). Its [DwC-A endpoint](http://www.gbif.es/ipt/archive.do?r=mzna_vert_mast_cas) is confirmed working. 

GBIF [automatically attempts to re-crawl each dataset every 7 days](https://github.com/gbif/ipt/wiki/FAQ.wiki#3-how-often-does-gbif-reindex-my-dataset). Unfortunately, this particular dataset has only ever been re-crawled 4 times since the date it was registered on Mar 22, 2016:
* Crawl attempt # 1 ran Mar 22, 2016, finishing successfully
* Crawl attempt # 2 ran Mar 22, 2016 finishing successfully but not modified
* Crawl attempt # 3 ran Mar 29, 2016, finishing successfully but not modified
* Crawl attempt # 4 is currently running.. 

Crawl attempt #4 is a phantom crawl having no start date and that has supposedly been running ever since the first week of April 2016. 

The consequence of this problem, is that GBIF failed to re-index the latest changes made to the dataset when it was republished on Apr 14, 2016. 

Additionally, it causes this dataset to be flagged as a false positive orphan dataset.

#### To sum up the problem
Phantom crawl attempts prevent datasets from being recrawled and leave stale data in GBIF's index. It also causes datasets to be erroneously flagged as orphaned datasets because they have failed to be re-crawled within several months.

#### Screenshots related to example above:
_Registry console showing crawl history and crawl running since April 2016_
![screen shot 2017-06-09 at 10 41 54](https://user-images.githubusercontent.com/5981100/26968900-1182b54a-4d04-11e7-95b0-7f6befc096a2.png)

_Properties of a phantom crawl:_
![screen shot 2017-06-09 at 10 17 49](https://user-images.githubusercontent.com/5981100/26968907-18fd8bce-4d04-11e7-9097-6b8cfa80672a.png)

**Comment by kbraak** [comment]
A .txt file with the UUIDs of 3466 datasets in this state can be downloaded [here](https://raw.githubusercontent.com/kbraak/watchdog/master/phantomCrawls-2017-06-14.txt). Thanks @timrobertson100 for offering to cleanup their state. It would be great if a recrawl could be initiated for these datasets, although for my purposes identifying orphans it's fine to skip recrawling datasets from [Pangae](http://www.gbif.org/publisher/d5778510-eb28-11da-8629-b8a03c50a862), [GEO-Tag der Artenvielfalt](http://www.gbif.org/publisher/ef69a030-3940-11dd-b168-b8a03c50a862). Thanks again.

**Comment by jlegind** [comment]
Many BioCASE datasets have not been crawled for months and I suspect it is because of the phantom crawls issue. Here are a few examples from the GBIF registry:

* https://registry.gbif.org/web/#/dataset/4c266400-b0a3-11e2-a01d-00145eb45e9a

**Comment by thomasstjerne** [comment]
There has been a couple of examples of datasets validating properly when submitted to the validator, but seemingly having indexing problems that could be related to this:

https://github.com/gbif/portal-feedback/issues/590
https://github.com/gbif/portal-feedback/issues/591

Also the crawler has ignored this dataset for a while and it is not pssoble to trigger the crawl from the registry or through the API:
https://www.gbif.org/dataset/2b94a042-fe01-4d9f-8995-d996c21d33cd

**Comment by MattBlissett** [comment]
I deleted the empty rows in Postgres, which has allowed the crawler scheduler to start a crawl on ~7000 datasets.

No new empty rows have appeared, but I'm going to see if it happens again before closing this.

```sql
select * from crawl_history where started_crawling is null order by started_crawling;
```

@gbif/content: Many hundreds of datasets have changes, but hadn't been ingested for (sometimes) over a year. I noticed at least one which duplicated itself. Also, it's not finished yet.

**Comment by MattBlissett** [comment]
One case where these appear is when crawler-clis are restarted, interrupting longrunning XML crawls, and those long-running crawls restarted. I'm not sure exactly which part caused the bad records.

The ideal method for stopping crawling is probably to stop the scheduler and coordinator, then wait until everything is finished before stopping the crawlers and fragmenters.

**Comment by MattBlissett** [comment]
The SQL above gives no more failures 1¾ years later, so I will close this.
